owncloudApp/src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.kt

```diff
@@ -0,0 +1,467 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.ui.fragment
+
+import android.accounts.Account
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.TextView
+import androidx.core.view.isVisible
+import com.owncloud.android.MainApp.Companion.appContext
+import com.owncloud.android.R
+import com.owncloud.android.datamodel.ThumbnailsCacheManager
+import com.owncloud.android.datamodel.ThumbnailsCacheManager.AsyncThumbnailDrawable
+import com.owncloud.android.datamodel.ThumbnailsCacheManager.ThumbnailGenerationTask
+import com.owncloud.android.domain.files.model.OCFile
+import com.owncloud.android.extensions.observeWorkerTillItFinishes
+import com.owncloud.android.extensions.showMessageInSnackbar
+import com.owncloud.android.files.FileMenuFilter
+import com.owncloud.android.files.services.FileUploader.FileUploaderBinder
+import com.owncloud.android.presentation.viewmodels.files.FileDetailsViewModel
+import com.owncloud.android.ui.activity.ComponentsGetter
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.ui.activity.FileDisplayActivity
+import com.owncloud.android.ui.controller.TransferProgressController
+import com.owncloud.android.ui.dialog.RemoveFilesDialogFragment
+import com.owncloud.android.ui.dialog.RenameFileDialogFragment
+import com.owncloud.android.utils.DisplayUtils
+import com.owncloud.android.utils.MimetypeIconUtil
+import com.owncloud.android.utils.PreferenceUtils
+import org.koin.android.ext.android.inject
+import timber.log.Timber
+
+/**
+ * This Fragment is used to display the details about a file.
+ */
+class FileDetailFragment : FileFragment() {
+    private var account: Account? = null
+    private var progressController: TransferProgressController? = null
+
+    private val fileDetailsViewModel: FileDetailsViewModel by inject()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        setHasOptionsMenu(true)
+
+        file = requireArguments().getParcelable(ARG_FILE)
+        account = requireArguments().getParcelable(ARG_ACCOUNT)
+        if (savedInstanceState != null) {
```

Comment:
`savedInstanceState?.let {`

Reply:
Not a huge difference, I think I will keep it 👍